go-controller/pkg/ovn/policy.go

```diff
@@ -665,16 +665,43 @@ func (oc *Controller) processLocalPodSelectorSetPods(policy *knet.NetworkPolicy,
 			continue
 		}
 
-		// Get the logical port info
 		logicalPort := util.GetLogicalPortName(pod.Namespace, pod.Name)
-		portInfo, err := oc.logicalPortCache.get(logicalPort)
-		// pod is not yet handled
-		// no big deal, we'll get the update when it is.
-		if err != nil {
+		var portInfo *lpInfo
+
+		// Get the logical port info from the cache, if that fails, retry
+		// if the gotten LSP is Scheduled for removal, retry (stateful-sets)
+		//
+		// 24ms is chosen because gomega.Eventually default timeout is 50ms
+		// libovsdb transactions take less than 50ms usually as well so pod create
+		// should be done within a couple iterations
+		retryErr := wait.PollImmediate(24*time.Millisecond, 120*time.Millisecond, func() (bool, error) {
```

Comment:
i'm wondering if 120 ms max is too short. Under heavy load it might take several hundred ms to add the logical port. I wonder if we should wait up to 500ms or something? maybe even a second? we are only stalling this specific network policy handler thread right? The trade off is i guess 1 pod could block several other pods from having their network policy applied for some length of time. Maybe we could get the LSPs in goroutines so we can do them all in parallel?

Reply:
Well the Local Pod Handler  `Addfunction` is called  individually for each pod matching a selector (https://github.com/ovn-org/ovn-kubernetes/blob/9e140b9c522a0ca085dc47c4e67ad4fd51b08741/go-controller/pkg/ovn/policy.go#L804) And I don't think each AddFunc call is blocking, so we _should_ be find just bumping the timeout... 

The other thing to remember here is that even if the first attempt times out, the pod should be taken care of during the update event that comes in when the ovn-k annotations are added to the pod right?

Reply:
So In that case I think bumping to 480 (20 retry cycles) wouldn't hurt here, but I don't think we should bump to a second, with the assumption that even if the first pod ADD doesn't set things up correctly the ensuing Pod UPDATE should